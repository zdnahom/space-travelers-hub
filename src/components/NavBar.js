import { NavLink } from 'react-router-dom';
import logo from '../assets/images/planet.png';

const NavBar = () => (
  <header>
    <nav>
      <img src={logo} alt="logo" className="logo" />
      <ul>
        <li><NavLink to="/">Rockets</NavLink></li>
        <li><NavLink to="/missions">Missions</NavLink></li>
        <li><NavLink to="/profile">My profile</NavLink></li>
      </ul>
    </nav>
  </header>
);
export default NavBar;
