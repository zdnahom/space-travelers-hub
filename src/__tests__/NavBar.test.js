import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import NavBar from '../components/NavBar';

afterEach(cleanup);
describe('Navbar component test', () => {
  it('It renders', () => {
    const { container } = render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
