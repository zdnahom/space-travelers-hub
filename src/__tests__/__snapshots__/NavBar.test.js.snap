// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component test It renders 1`] = `
<div>
  <header
    class="navbar-container"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo-container"
      >
        <img
          alt="logo"
          class="logo"
          src="planet.png"
        />
        <span
          class="logo-text"
        >
          Space Travelers' Hub
        </span>
      </div>
      <ul
        class="navbar-ul"
      >
        <li>
          <a
            aria-current="page"
            class="active-link"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="vertical-line"
        />
        <li>
          <a
            class="nav-link"
            href="/profile"
          >
            My profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
